### DISCUSSION TITLE
Is purego.Dlsym supported on Windows?

### DISCUSSION THREAD

**Discussion by texadactyl** [ebitengine/purego#280] [General]
I get compilation errors on Windows: `undefined purego.Dlsym`.

Said before in a different discussion: Please update https://pkg.go.dev/github.com/ebitengine/purego to indicate which functions are supported on Windows and which are supported on POSIX systems. Thank you.

cc: @platypusguy

**Comment by TotallyGamerJet** [comment]
The [examples/libc](https://github.com/ebitengine/purego/tree/main/examples/libc) shows how to use purego on Windows.

The comments were updated but a release hasn't been made yet so it doesn’t show up in go doc. https://github.com/ebitengine/purego/commit/2d5afc8a4b0c49b48c2b6c37dad1512cb24c28a4

**Comment by hajimehoshi** [comment]
You can switch GOOS for documents

https://pkg.go.dev/github.com/ebitengine/purego?GOOS=linux
https://pkg.go.dev/github.com/ebitengine/purego?GOOS=windows

**Reply by texadactyl** [reply]
Thank you. That helped me considerably.

Under "Supported Platforms" or under "Documentation", you should state explicitly when there are functions that are applicable by the user's resident O/S. It's too easy to miss "Rendered for".  

However, I still think that *all* API documentation (not just yours) should indicate under each function if there are any O/S restrictions. Excellent example: https://pkg.go.dev/os clearly specifies under each function restrictions by O/S to the function or the data in use or produced by the function.

**Reply by hajimehoshi** [reply]
As @TotallyGamerJet mentioned, we have commited 2d5afc8a4b0c49b48c2b6c37dad1512cb24c28a4 to add more explanations about Windows for Dl* functions. Please stay tuned for the next stable release. Thanks!
